specification/ai/Azure.AI.Projects/servicepatterns.tsp

```diff
@@ -0,0 +1,86 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+
+namespace Azure.AI.Projects.ServicePatterns {
+  @Rest.autoRoute
+  interface VersionedOperations<
+    TEntityType extends Reflection.Model,
+    TListLatestParameters extends Reflection.Model = {},
+    TListVersionsParameters extends Reflection.Model = {}
+  > {
+    @Rest.action("versions")
+    @Rest.actionSeparator("/")
+    @Http.get
+    listVersions is BuildingBlocks.CoreOps.ResourceAction<
+      TEntityType,
+      TListVersionsParameters,
+      Azure.Core.Page<TEntityType>
+    >;
+
+    @Rest.listsResource(TEntityType)
+    listLatest is Azure.Core.Foundations.ResourceList<
+      TEntityType,
+      TListLatestParameters,
+      Azure.Core.Page<TEntityType>
+    >;
+
+    getVersion is Azure.Core.Foundations.ResourceOperation<
+      TEntityType,
+      {
+        @Rest.segment("versions")
+        @Http.path
+        version: string;
+      },
+      Azure.Core.Foundations.ResourceOkResponse<TEntityType>
+    >;
+
+    @Http.delete
+    deleteVersion is Azure.Core.Foundations.ResourceOperation<
+      TEntityType,
+      {
+        @Rest.segment("versions")
+        @Http.path
+        version: string;
+      },
+      Http.NoContentResponse
+    >;
+
+    create is BuildingBlocks.RepeatableCoreOps.ResourceCreateWithServiceProvidedName<TEntityType>;
```

Comment:
https://github.com/azure-sdk/azure-sdk-for-python/blob/49579936821bc59472c092ca426fc882afa410a6/sdk/ai/azure-ai-projects-dp1/azure/ai/projects/dp1/operations/_operations.py#L7466C5-L7468C15

Create should still require a `name`. It's just the version that's not required and will be autogenerated.